x-pack/platform/plugins/shared/sample_data_ingest/server/services/artifact_manager/utils/resolve_local_path.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export function resolveLocalPath(parsedUrl: URL): string {
```

Comment:
From a security standpoint I think we're trying to get away from any filesystem access so this is a bit of a red flag for me.